### PR TITLE
clientv3: add auth example

### PR DESCRIPTION
**Pull request by gyuho**

### CODE REVIEW

**Comment by gyuho** [comment]
/cc @mitake @xiang90 @heyitsanthony 

Thanks.

File: clientv3/example_auth_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		log.Fatal(err)
+	}
+
+	_, err = kv.Txn(context.TODO()).
+		If(clientv3.Compare(clientv3.Value("zoo1"), ">", "abc")).
+		Then(clientv3.OpPut("zoo1", "XYZ")).
+		Else(clientv3.OpPut("zoo1", "ABC")).
+		Commit()
+	fmt.Println(err)
+	// Output: etcdserver: permission denied
```
Comment (xiang90) [inline clientv3/example_auth_test.go]: probably we also want to show how to pass the permission checking by doing authentication and pass in username?

Comment (gyuho) [inline clientv3/example_auth_test.go]: Sure, will add. thanks.

**Comment by gyuho** [comment]
@xiang90 Just added. PTAL.

Thanks!

**Comment by gyuho** [comment]
Had to add AuthDisable for other tests

```
=== RUN   ExampleKV_get
2016-06-23 03:00:04.487152 E | auth: invalid user name  for permission checking
2016-06-23 03:00:04.487765 E | auth: invalid user name  for permission checking
2016-06-23 03:00:04.487901 I | etcdserver: permission denied
FAIL	github.com/coreos/etcd/clientv3	5.322s
```

Now tests fixed.

**Comment by xiang90** [comment]
> 2016-06-23 03:00:04.487152 E | auth: invalid user name  for permission checking
> 2016-06-23 03:00:04.487765 E | auth: invalid user name  for permission checking

Why do we see these errors?

**Comment by gyuho** [comment]
@xiang90 `ExampleKV_get` shares the cluster used in `ExampleAuth`, and my previous code wasn't disabling auth, so it was running `ExampleKV_get` with auth enabled, but `ExampleKV_get` client wasn't using any auth, so those errors. Now we don't see those errors.

**Comment by xiang90** [comment]
OK. LGTM then.

**Comment by gyuho** [comment]
thanks.

**Comment by mitake** [comment]
@gyuho thanks!

**Pull request merged**